deltacat/tests/compute/compact_partition_rebase_then_incremental_test_cases.py

```diff
@@ -0,0 +1,1533 @@
+            (
+                pa.Table.from_arrays(
+                    [
+                        pa.array([0]),
+                        pa.array(["1"]),
+                    ],
+                    names=["pk_col_1", "col_1"],
+                ),
+                DeltaType.UPSERT,
+                None,
+            ),
+            (
+                pa.Table.from_arrays(
+                    [
+                        pa.array(["1"]),
+                    ],
+                    names=["col_1"],
+                ),
+                DeltaType.DELETE,
+                None,
+            ),
+        ],
+        expected_terminal_compact_partition_result=pa.Table.from_arrays(
+            [
+                pa.array([]),
+                pa.array([]),
+            ],
+            schema=pa.schema(
+                [
+                    ("pk_col_1", pa.int64()),
+                    ("col_1", pa.string()),
+                ]
+            ),
+        ),
+        expected_terminal_exception=AssertionError,
+        do_create_placement_group=True,
+        records_per_compacted_file=DEFAULT_MAX_RECORDS_PER_FILE,
+        hash_bucket_count=DEFAULT_HASH_BUCKET_COUNT,
+        read_kwargs_provider=None,
+        drop_duplicates=True,
+        skip_enabled_compact_partition_drivers=[CompactorVersion.V1],
+    ),
+    "24-rebase-then-incremental-delete-type-delta-delete-column-no-records": RebaseThenIncrementalCompactionTestCaseParams(
+        primary_keys={"pk_col_1"},
+        sort_keys=ZERO_VALUED_SORT_KEY,
+        partition_keys=[PartitionKey.of("region_id", PartitionKeyType.TIMESTAMP)],
+        partition_values=["2022-01-01T00:00:00.000Z"],
+        input_deltas=pa.Table.from_arrays(
+            [
+                pa.array([0, 1]),
+                pa.array(["0", "1"]),
+            ],
+            names=["pk_col_1", "col_1"],
+        ),
+        input_deltas_delta_type=DeltaType.UPSERT,
+        rebase_expected_compact_partition_result=pa.Table.from_arrays(
+            [
+                pa.array([0, 1]),
+                pa.array(["0", "1"]),
+            ],
+            names=["pk_col_1", "col_1"],
+        ),
+        incremental_deltas=[
+            (
+                pa.Table.from_arrays(
+                    [
+                        pa.array([0]),
+                        pa.array(["1"]),
+                    ],
+                    names=["pk_col_1", "col_1"],
+                ),
+                DeltaType.UPSERT,
+                None,
+            ),
+            (
+                pa.Table.from_arrays(
+                    [
+                        pa.array([]),
+                        pa.array([]),
+                    ],
+                    schema=pa.schema(
+                        [
+                            ("pk_col_1", pa.int64()),
+                            ("col_1", pa.string()),
+                        ]
+                    ),
+                ),
+                DeltaType.DELETE,
+                None,
```

Comment:
When delete column is present but not records should succeed without changing the table.